src/commands.ts

```diff
@@ -0,0 +1,168 @@
+    } catch (err) {
+        vscode.window.showErrorMessage(err);
+    }
+}
+
+async function validateProjectSDK(projects: string[], commandObserver: CommandObserver): Promise<boolean> {
+    var packageInfo = Utils.getPackageInfo();
+    let unsupportedProjects = await Helper.checkProjectVersion(packageInfo.minSupportedPostgreSQLProjectSDK, packageInfo.maxSupportedPostgreSQLProjectSDK, projects, commandObserver);
+    if (unsupportedProjects && unsupportedProjects.length > 0) {
+        unsupportedProjects.map(p => commandObserver.logToOutputChannel(strings.format(Constants.buildFailedUnsupportedSdkMessage, p)));
+        return Promise.resolve(false);
+    }
+    return Promise.resolve(true);
+}
+
+async function dotnetBuild(dotNetSdk: DotNetInfo, project: string, commandObserver: CommandObserver, cancelToken: vscode.CancellationToken): Promise<void> {
+    let args = ['build', project];
+    commandObserver.logToOutputChannel(strings.format(Constants.buildStartedMessage, project));
+    await runDotNetCommand(dotNetSdk, args, commandObserver, cancelToken);
+    commandObserver.logToOutputChannel(strings.format(Constants.buildCompletedMessage, project));
+}
+
+async function addNewPostgreSQLProject(args: vscode.Uri, dotNetSdk: DotNetInfo, projectTemplateNugedId: string, commandObserver: CommandObserver) {
+    let folder = args.fsPath;
+    var defaultProjectName = path.basename(folder);
+
+    var projectName = await vscode.window.showInputBox(
+        {
+            prompt: 'Project Name',
+            value: defaultProjectName,
+            validateInput: (value: string) => {
+                if (!value || value.trim().length === 0) {
+                    return Constants.projectNameEmptyErrorMessage;
+                }
+                if (/[\/?:&*"<>|#%;\\]/g.test(value))
+                {
+                    return Constants.projectNameSpecialCharsErrorMessage;
+                }
+                if (value === '.' || value === '..')
+                {
+                    return Constants.projectNameInvalidErrorMessage;
+                }
+                if (value.includes('..'))
+                {
+                    return Constants.projectNameInvalidCharErrorMessage;
+                }
+
+                return null;
+            }
+        });
+
+    if (projectName === undefined) {
+        return;
+    }
+
+    findProjectTemplate(dotNetSdk).then(exists => {
+        commandObserver.clear();
+        if (exists) {
+            dotnetNew(dotNetSdk, projectName, folder, commandObserver);
+        }
+        else {
+            installPostgreSQLProjectTemplate(dotNetSdk, projectTemplateNugedId, commandObserver).then(() => {
+                dotnetNew(dotNetSdk, projectName, folder, commandObserver);
+            });
+        }
+    });
+}
+
+function installPostgreSQLProjectTemplate(dotNetSdk: DotNetInfo, templateNupkg: string, commandObserver: CommandObserver): Promise<void> {
+	return new Promise<void>((_resolve, _reject) => {
```

Comment:
Fix indentation to use 4 spaces instead of tabs

Reply:
They have been using tabs everywhere for ts files. Even is ADS codebase. 
Shouldn't we keep it consistent?